addon/mixins/ajax-request.js

```diff
@@ -42,16 +42,13 @@ const JSONAPIContentType = 'application/vnd.api+json';
 // Header names are case-insensitive
 // https://www.w3.org/Protocols/rfc2616/rfc2616.html
 function retrieveHeader(headers, name) {
-  let found = null;
-  
+  var found = null;
```

Comment:
Gotcha.  All of that is a little too stateful for my taste.  The approach I took is as follows, but feel free to do it however you like:

```js
  const matchingKey = Ember.A(Object.keys(headers)).find((key) => {
    return key.toLowerCase() === desiredKey.toLowerCase();
  });
  if (matchingKey) {
    return headers[matchingKey];
  }
```